lib/Sema/TypeCheckGeneric.cpp

```diff
@@ -134,34 +136,45 @@ OpaqueResultTypeRequest::evaluate(Evaluator &evaluator,
       return nullptr;
     }
   } else {
-    opaqueReprs = repr->collectOpaqueReturnTypeReprs();
+      if (ctx.LangOpts.hasFeature(Feature::ImplicitSome)) {
+          opaqueReprs = collectTypeReprs(repr);
+      } else {
+          opaqueReprs = collectOpaqueReturnTypeReprs(repr);
+      }
     SmallVector<GenericTypeParamType *, 2> genericParamTypes;
     SmallVector<Requirement, 2> requirements;
     for (unsigned i = 0; i < opaqueReprs.size(); ++i) {
       auto *currentRepr = opaqueReprs[i];
+        
+      if (isa<IdentTypeRepr>(currentRepr)){
+        if(!isProtocol(evaluator, currentRepr, ctx, dc))
+          continue;
+      }
 
+      if( auto opaqueReturnRepr = dyn_cast<OpaqueReturnTypeRepr>(currentRepr) ) {
       // Usually, we resolve the opaque constraint and bail if it isn't a class
       // or existential type (see below). However, in this case we know we will
       // fail, so we can bail early and provide a better diagnostic.
       if (auto *optionalRepr =
-              dyn_cast<OptionalTypeRepr>(currentRepr->getConstraint())) {
+              dyn_cast<OptionalTypeRepr>(opaqueReturnRepr->getConstraint())) {
         std::string buf;
         llvm::raw_string_ostream stream(buf);
         stream << "(some " << optionalRepr->getBase() << ")?";
 
         ctx.Diags.diagnose(currentRepr->getLoc(),
                            diag::opaque_type_invalid_constraint);
         ctx.Diags
-            .diagnose(currentRepr->getLoc(), diag::opaque_of_optional_rewrite)
-            .fixItReplaceChars(currentRepr->getStartLoc(),
-                               currentRepr->getEndLoc(), stream.str());
+           .diagnose(currentRepr->getLoc(), diag::opaque_of_optional_rewrite)
+           .fixItReplaceChars(currentRepr->getStartLoc(),
+                              currentRepr->getEndLoc(), stream.str());
         return nullptr;
-      }
+      }}
```

Comment:
Nit: Add the new closing brace on the next line. We should also indent the code inside your new condition (from 159-171) two spaces.